utils.js

```diff
@@ -0,0 +1,30 @@
+/**
+ * 
+ * @param {String} hex 
+ * @returns {String}
+ */
+function fromHexToText(hex){
+    const every2Chars = /.{2}/g;
+    const match = hex.match(every2Chars);
+    if(match) {
+        return match.map(
+            // converts each hex character to text
+            (v) => String.fromCharCode(parseInt(v,16))
+        ).join("");
+    }else{
+        return "";
+    }
+}
```

Comment:
Unsure if the function above is the best way to get to the text version of an asset name 🤔

Reply:
You could have a look at the `bytesToString` function in the helios library itself (it might be smart to export it at some point so we wouldn't need to reimplement it in PicoSwap). So it would be something like:
```
function hexToString(hex) {
    try {   
        return (new TextDecoder("utf-8", {fatal: true})).decode((new Uint8Array(helios.hexToBytes(hex))).buffer);
    } catch (e) {
        return hex // fallback in tokenName isn't valid UTF-8 (which is allowed according to Plutus Ledger API)
    }
}
```

I would also fall back to original hex if it isn't valid UTF-8

Reply:
Thanks. Submitted a [PR to helios](https://github.com/Hyperion-BT/Helios/pull/27) with the suggested change. Once that's in place, I believe the final version of the code will look something like this:


```typescript
import { hexToBytes, bytesToString } from "./helios.js";

function hexToString(hex){
    try {
        return bytesToString(hexToBytes(hex));
    } catch (e) {
        return hex;
    }
}
```